publish.sh

```diff
@@ -2,35 +2,92 @@
 
 set -ex
 
+readonly ARTIFACTS=${ARTIFACTS:-${PWD}}
+readonly CONCURRENT_PROVISION_JOBS_COUNT="${CONCURRENT_PROVISION_JOBS_COUNT:-3}"
+
 KUBEVIRTCI_TAG=$(date +"%y%m%d%H%M")-$(git rev-parse --short HEAD)
 export KUBEVIRTCI_TAG
 
 TARGET_REPO="quay.io/kubevirtci"
 TARGET_KUBEVIRT_REPO="quay.io/kubevirt"
 TARGET_GIT_REMOTE="https://kubevirt-bot@github.com/kubevirt/kubevirtci.git"
 
+exec_on_background() {
+  local -r job_name_prefix=$1
+  local -r jobs=($2)
+  local -r command=$3
+
+  local -r jobs_count="${#jobs[@]}"
+  local offset=0
+  while [ "${jobs_count}" -gt "${offset}" ]; do
+      jobs_batch_pids=()
+      jobs_batch=("${jobs[@]:${offset}:${CONCURRENT_PROVISION_JOBS_COUNT}}")
+      for job in ${jobs_batch[@]}; do
+          eval "${command}" &> "${ARTIFACTS}/${job_name_prefix}-${job}.log" &
+          jobs_batch_pids+=($!)
+          echo "[$(date --utc)] ${job_name_prefix} started ${job}, PID: $!.."
+      done
+      offset=$((offset+CONCURRENT_PROVISION_JOBS_COUNT))
+
+      # wait for jobs to finish
+      set +e
+      for job_pid in "${jobs_batch_pids[@]}"; do
+          echo "[$(date --utc)] waiting for ${job_name_prefix} job, PID: ${job_pid}.."
+          wait ${job_pid}
```

Comment:
you are waiting serially, better to wait to any that occur
so if it fails, you fail fast